error_check_logic.py

```diff
@@ -12,229 +13,268 @@
-  result = Validate(institution, session, models.GroupsStudents, schemas.student_groups)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\nValidate Student Groups: ' + result
-  result = CheckClassDayparts(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class dayparts against Dayparts: ' + result
-  result = CheckDaypartsUsed(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck if all Dayparts are used: ' + result
-  result = CheckClassStudentGroups(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class student groups against Student Groups: ' + result
-  result = CheckStudentGroupsUsed(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck if all Student Groups are used: ' + result
-  result = CheckClassEmails(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class student emails against Students: ' + result
-  result = CheckCurrentGrade(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class current_grade against Student current_grade: ' + result
-  return setup_msg, error_chk
+class Checker(object):
+  def __init__(self):
+    self.error_check_msg = 'OK' # setup_msg
+    self.error_check_detail = '' # error_chk
+
+  # Modules that change admin setup data should call this
+  # with status = 'UNKNOWN'.
+  @classmethod
+  def setStatus(self, institution, session, status = 'UNKNOWN'):
+    models.ErrorCheck.Store(institution, session, status)
+
+  @classmethod
+  def getStatus(self, institution, session):
+    return models.ErrorCheck.Fetch(institution, session)
+
+  # Returns two values:
+  #   error_check_msg: 'OK' if all tests pass
+  #                    'FAIL' if any single test fails
+  #   error_check_detail: String containing detailed results
+  #                       for ALL tests.
+  # Sets datastore to value of error_check_msg
+  def Run(self, institution, session):
+    dayparts = models.Dayparts.Fetch(institution, session)
+    classes = models.Classes.Fetch(institution, session)
+    students = models.Students.Fetch(institution, session)
+    requirements = models.Requirements.Fetch(institution, session)
+    class_groups = models.GroupsClasses.Fetch(institution, session)
+    student_groups = models.GroupsStudents.Fetch(institution, session)
+
+    valid_dayparts = schemas.dayparts.IsValid(dayparts)
+    valid_classes = schemas.classes.IsValid(classes)
+    valid_students = schemas.students.IsValid(students)
+    valid_requirements = schemas.requirements.IsValid(requirements)
+    valid_class_groups = schemas.class_groups.IsValid(class_groups)
+    valid_student_groups = schemas.student_groups.IsValid(student_groups)
+
+    self.error_check_detail += '\nValidate Dayparts: ' +\
+      self._Validate(dayparts, valid_dayparts)
```

Comment:
You could pass the string "Dayparts" in to the _Validate method and then build the error_check_detail string inside the method:
```python
self._Validate(dayparts, valid_dayparts, "Dayparts")
```